### ISSUE TITLE
"Parameter/variable not used" false-positive (ArrayAccess inherited objects)

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#614]
Consider the following code.

```php
$foo = new Foo($bar = new Bar);
$bar['baz'] = 'qux';
```

The inspection claims that parameter/variable `$bar['baz']` is overridden but never used. However, since this is an assignment expression mutating `$bar` and `$bar` has already been passed to `Foo`, the modification affects the instance of `Bar` assigned to `Foo` and is therefore not, as the inspection implies, dead code.

**Comment by kalessil** [comment]
@Bilge : not completely broken =) But definitely this is a false-positive

**Comment by Bilge** [comment]
👏
